code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -214,122 +214,129 @@
+		if(cell)
 			balloon_alert(user, "already installed!")
-		return
+			return ITEM_INTERACT_BLOCKING
 
-	if(istype(weapon, /obj/item/stock_parts/scanning_module))
-		if(!scanmod)
-			if(!user.transferItemToLoc(weapon, src, silent = FALSE))
-				return
-			scanmod = weapon
-			balloon_alert(user, "installed scanning module")
-			playsound(src, 'sound/items/tools/screwdriver2.ogg', 50, FALSE)
-			log_message("[weapon] installed", LOG_MECHA)
-			update_part_values()
-		else
+		if(!user.transferItemToLoc(tool, src, silent = FALSE))
+			return  ITEM_INTERACT_BLOCKING
+
+		cell = tool
+		balloon_alert(user, "installed power cell")
+		diag_hud_set_mechcell()
+		playsound(src, 'sound/items/tools/screwdriver2.ogg', 50, FALSE)
+		log_message("Power cell installed", LOG_MECHA)
+		return ITEM_INTERACT_SUCCESS
+
+	if(istype(tool, /obj/item/stock_parts/scanning_module))
+		if(scanmod)
 			balloon_alert(user, "already installed!")
-		return
+			return ITEM_INTERACT_BLOCKING
 
-	if(istype(weapon, /obj/item/stock_parts/capacitor))
-		if(!capacitor)
-			if(!user.transferItemToLoc(weapon, src, silent = FALSE))
-				return
-			capacitor = weapon
-			balloon_alert(user, "installed capacitor")
-			playsound(src, 'sound/items/tools/screwdriver2.ogg', 50, FALSE)
-			log_message("[weapon] installed", LOG_MECHA)
-			update_part_values()
-		else
+		if(!user.transferItemToLoc(tool, src, silent = FALSE))
+			return ITEM_INTERACT_BLOCKING
+
+		scanmod = tool
+		balloon_alert(user, "installed scanning module")
+		playsound(src, 'sound/items/tools/screwdriver2.ogg', 50, FALSE)
+		log_message("[tool] installed", LOG_MECHA)
+		update_part_values()
+		return ITEM_INTERACT_SUCCESS
+
+	if(istype(tool, /obj/item/stock_parts/capacitor))
+		if(capacitor)
 			balloon_alert(user, "already installed!")
-		return
+			return ITEM_INTERACT_BLOCKING
 
-	if(istype(weapon, /obj/item/stock_parts/servo))
-		if(!servo)
-			if(!user.transferItemToLoc(weapon, src, silent = FALSE))
-				return
-			servo = weapon
-			balloon_alert(user, "installed servo")
-			playsound(src, 'sound/items/tools/screwdriver2.ogg', 50, FALSE)
-			log_message("[weapon] installed", LOG_MECHA)
-			update_part_values()
-		else
+		if(!user.transferItemToLoc(tool, src, silent = FALSE))
+			return ITEM_INTERACT_BLOCKING
+
+		capacitor = tool
+		balloon_alert(user, "installed capacitor")
+		playsound(src, 'sound/items/tools/screwdriver2.ogg', 50, FALSE)
+		log_message("[tool] installed", LOG_MECHA)
+		update_part_values()
+		return ITEM_INTERACT_SUCCESS
+
+	if(istype(tool, /obj/item/stock_parts/servo))
+		if(servo)
 			balloon_alert(user, "already installed!")
-		return
+			return ITEM_INTERACT_BLOCKING
+
+		if(!user.transferItemToLoc(tool, src, silent = FALSE))
+			return ITEM_INTERACT_BLOCKING
+
+		servo = tool
+		balloon_alert(user, "installed servo")
+		playsound(src, 'sound/items/tools/screwdriver2.ogg', 50, FALSE)
+		log_message("[tool] installed", LOG_MECHA)
+		update_part_values()
```

Comment:
```suggestion
		update_part_values()
		return ITEM_INTERACT_SUCCESS
```

Reply:
I'm not sure if I understand, I added the return. If you mean at the end of the block of code, that's intended or else it'll cancel the attack chain.